lib/upload.js

```diff
@@ -191,27 +199,33 @@ class Upload {
     }
 
     // Try to find the endpoint for the file in the storage
-    if (this.options.resume) {
+    if (this.options.resume && this._storage) {
```

Comment:
`this.options.resume && this._storage` is repeated quite often. Please use a private property such as `this._hasURLStorage` instead.